LightFramework/src/rendering/scenerenderer.cpp

```diff
@@ -124,31 +164,49 @@ namespace Light {
 
 		m_framebuffer->clearAttachment(1, 0);
 
-		Light::Renderer::beginScene(camera, camera.getViewMatrix());
+		Light::Renderer::beginScene(camera, camera.getViewMatrix(), camera.calculatePosition());
 
-		std::vector<PointLight> lights;
+		std::vector<PointLight> pointLights;
+		std::vector<SpotLight> spotLights;
+		std::vector<DirectionalLight> directionalLights;
 		{
 			auto view = scene->m_registry.view<LightComponent, TransformComponent>();
-			for(auto& entity: view)
+
+			for (auto &entity : view)
 			{
-				auto[light, transform] = view.get(entity);
-				lights.push_back({transform.position, light.m_lightColor});
+
+				auto [light, transform] = view.get(entity);
+				switch ((int)light.m_lightType)
```

Comment:
Is this type cast really necessary? If it is, consider using a C++ style typecast instead.

Reply:
Yeah, it is.  I have changed it into a C++ static_cast